spirv_msl.cpp

```diff
@@ -1464,6 +1466,27 @@ void CompilerMSL::extract_global_variables_from_function(uint32_t func_id, std::
 				break;
 			}
 
+			case OpExtInst:
+			{
+				uint32_t extension_set = ops[2];
+				if (get<SPIRExtension>(extension_set).ext == SPIRExtension::GLSL)
+				{
+					auto op_450 = static_cast<GLSLstd450>(ops[3]);
+					switch (op_450)
+					{
+					case GLSLstd450InterpolateAtCentroid:
+					case GLSLstd450InterpolateAtSample:
+					case GLSLstd450InterpolateAtOffset:
+					{
+						// For these, we really need the stage-in block.
+						added_arg_ids.insert(stage_in_var_id);
```

Comment:
Fallthrough. Benign, but looks weird.

Reply:
Also, why stage_in_var_id and not the variable it touches itself?

Reply:
> Also, why stage_in_var_id and not the variable it pulls from itself?

I considered that, but supporting composites with that seemed really complicated.

Reply:
> > Also, why stage_in_var_id and not the variable it pulls from itself?
> 
> I considered that, but supporting composites with that seemed really complicated.

I've added a comment explaining why I did it this way.